### PR TITLE
Added size field in interfaces

### PR DESCRIPTION
**Pull request by rahuyada**
Added size field in interfaces which will check if enough memory is allocated to hold requested information.

Signed-off-by: Rahul Yadav <rahul.x.yadav@oracle.com>

### CODE REVIEW

File: resource.c [hunk 1]
```diff
@@ -231,17 +239,23 @@ int res_read_blk(res_blk_t *res, int pid, int flags)
...
 		case RES_KERN_RELEASE:
 			ret = uname(&t);
 			if (ret == -1) {
 				res->res_unit[i]->status = errno;
 			} else {
-				strncpy((res->res_unit[i]->data).str,
-					t.release, RESOURCE_64);
+				out = (res->res_unit[i]->data).str;
+				len = sizeof(union r_data);
+				strncpy(out, t.release, len-1);
```
Comment (hallyn) [inline resource.c]: In this case this probably can't happen, but as a code pattern goes, the problem here is that if len > length of the string in t.release, then you may leak uninitialized memory here.

File: resource.c [hunk 2]
```diff
@@ -194,7 +198,9 @@ int res_read(int res_id, void *out, void *hint, int pid, int flags)
 			errno = err;
 			return -1;
 		}
-		sscanf(t.version, "%*s%*s%*s%[^\t\n]", (char *) out);
+		sscanf(t.version, "%*s%*s%*s%[^\t\n]", t.version);
```
Comment (hallyn) [inline resource.c]: sscanf into t.version from t.version is strange.  Not sure offhand whteher sscanf guarantees that it won't overwrite it's working space.

File: resnet.c [hunk 3]
```diff
@@ -258,12 +258,22 @@ int getnetinfo(int res_id, void *out, void *hint, int pid, int flags)
...
+	}
+
 	switch (res_id) {
 	case RES_NET_IFSTAT:
+		CHECK_SIZE(sz, sizeof(res_net_ifstat_t));
+
 		/* Interface name should be provided */
 		if (hint == NULL) {
 			eprintf(
-				"Interface name is not provided for RES_NET_IFSTAT");
```
Comment (hallyn) [inline resnet.c]: Just curious, why did you drop this?

**Comment by hallyn** [comment]
Overall +1 on the idea.

File: resnet.c (hunk 3 above)
Comment (rahuyada) [reply resnet.c]: I was thinking I don't need to provide resource ID name as application will have it while calling.
But keeping it makes it easier for application so putting this back with line length fix.

File: resource.c (hunk 2 above)
Comment (rahuyada) [reply resource.c]: It worked, however this does not look clean, so I have changed this with a cleaner solution.

**Comment by rahuyada** [comment]
@brauner Please let me know if you have any further comments.
If you are ok with the PR please merge it.

**Pull request merged**